### DISCUSSION TITLE
Is it possible to specify content-encoding headers for static assets in the parcel dev server?

### DISCUSSION THREAD

**Discussion by jeffpeck10x** [parcel-bundler/parcel#5778] [Q&A] [answered]
I have a project that includes some sample data for a demo. That data is gzipped and is to be loaded by the application at runtime.

I would like to _somehow_ specify that the http response headers for this content should include `content-encoding: gzip`. Is that possible with a parcel config, (maybe something in .parcelrc)?

There are several reasonable workarounds for my use-case, so this is not a big deal, but I am curious if there is a "correct" way to handle this.

**Comment by mischnic** [comment, marked as answer]
You should be able to set that with a `.proxyrc.js`: https://v2.parceljs.org/features/api-proxy/#.proxyrc.js

**Reply by zanona** [reply]
@mischnic, how would you go about only adding custom http headers to the response? _(my use case is replicating CSP headers I have on the production server)_

.proxyrc.json
```json
{"/": {"target": "http://localhost:1234", "headers": {"Content-Security-Policy": "..."}}}
```
or  with .proxyrc.js
```js
app.use(createProxyMiddleware("/", {target: "http://localhost:1234", headers: {"Content-Security-Policy": "..."}}));
```
would trigger an infinite loop as it's proxying http://localhost:1234 to the same target?

**Reply by zanona** [reply]
Sorry, Never mind :) I have just noticed we can create another middleware from the exported module function giving us complete control over the response.

```js
module.exports = function (app) {
	app.use((req, res, next) => {
		const ext = extname(req.url);
		if (ext === "" || ext === ".html") {
			res.setHeader('Content-Security-Policy', '...');
		}
		next();
	});
        // other proxied requests 
        app.use(createProxyMiddleware('/api/*', {...}))
}
```
